pkg/controllers/nodeclaim/inplaceupdate/patch.go

```diff
@@ -133,3 +180,87 @@ func patchVMTags(
 	update.Tags = expectedTags
 	return true
 }
+
+func patchAKSMachineTags(
+	update *armcontainerservice.Machine,
+	params *patchParameters,
+	patchingAKSMachine *armcontainerservice.Machine,
+) bool {
+	creationTimestamp := getCorrectedAKSMachineCreationTimestamp(patchingAKSMachine)
+	// For NodeClaim name tag, given this controller is based on actual NodeClaim like during Create(), the patch will repair the tag if needed.
+
+	expectedTags := instance.ConfigureAKSMachineTags(
+		params.opts,
+		params.nodeClass,
+		params.nodeClaim,
+		creationTimestamp,
+	)
+
+	eq := func(v1, v2 *string) bool {
```

Comment:
minor: maybe we should define this as a local function so we can reuse it between `patchVMTags` and this?